### PR TITLE
SNOW-1947479 Add bulk_upload_chunks parameter to write_pandas

### PR DESCRIPTION
**Pull request by sfc-gh-mkubik**
Add `bulk_upload_chunks` parameter to `write_pandas` function to enable upload of all chunks at once.

### CODE REVIEW

File: test/integ/pandas/test_pandas_tools.py [hunk 1]
```diff
@@ -1139,3 +1139,49 @@ def test_pandas_with_single_quote(
...
+    random_table_name = random_string(5, "userspoints_")
+    df_data = [("Dash", 50), ("Luke", 20), ("Mark", 10), ("John", 30)]
+    df = pandas.DataFrame(df_data, columns=["name", "points"])
+
+    table_name = random_table_name
+    col_id = "id"
+    col_name = "name"
+    col_points = "points"
+
+    create_sql = (
```
Comment (sfc-gh-fpawlowski) [inline test/integ/pandas/test_pandas_tools.py:1156]: nit: could be nice to have a context-manager fixture (maybe factory) that would create table and drop it after leaving the context - it is quite a common thing to do in this tests file.

But it could be considered in the future enhancements as well.

**Review by sfc-gh-fpawlowski — CHANGES_REQUESTED** [review]

File: src/snowflake/connector/pandas_tools.py [hunk 2]
```diff
@@ -437,17 +440,27 @@ def write_pandas(
...
+                # Upload parquet file chunk right away
+                path = chunk_path.replace("\\", "\\\\").replace("'", "\\'")
+                cursor._upload(
+                    local_file_name=f"'file://{path}'",
+                    stage_location="@" + stage_location,
+                    options={"parallel": parallel, "source_compression": "auto_detect"},
+                )
+
+                # Remove chunk file
+                os.remove(chunk_path)
```
[the remark this answers is not in the document]
Comment (sfc-gh-mkubik) [reply src/snowflake/connector/pandas_tools.py:453]: That's the only reason that I can think of. Sounds like an edge case so possibly we'll be able to make the aggregated upload a default behaviour

File: test/integ/pandas/test_pandas_tools.py [hunk 3]
```diff
@@ -1139,3 +1139,49 @@ def test_pandas_with_single_quote(
             )
         finally:
             cnx.execute_string(f"drop table if exists {table_name}")
+
+
+@pytest.mark.parametrize("write_then_upload", [True, False])
+def test_write_pandas_wildcard_upload(conn_cnx, write_then_upload):
```
[the remark this answers is not in the document]
Comment (sfc-gh-mkubik) [reply test/integ/pandas/test_pandas_tools.py:1145]: That's the previous name I've used for this parameter, good catch

File: test/integ/pandas/test_pandas_tools.py (hunk 1 above)
Comment (sfc-gh-mkubik) [reply test/integ/pandas/test_pandas_tools.py:1156]: actually in `test/integ/pandas/test_pandas_tools.py` it seems that the conventions is to drop the test table in finally block but it makes sense to unify this file with other tests. I'd do a separate PR for that though.

Comment (sfc-gh-fpawlowski) [reply test/integ/pandas/test_pandas_tools.py:1156]: Yes - that the whole point - the context manager would contain that exact 'finally' block and execute it automatically as fixture (or fixture-produced ContextManager) exits the context of 'with' block.

But I agree - this can be done in a separate PR

**Review by sfc-gh-pbulawa — APPROVED** [review]
LGTM

**Review by sfc-gh-fpawlowski — APPROVED** [review]
